test/bench/synthetic_benchmarks.cpp

```diff
@@ -0,0 +1,241 @@
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * (bytecode{opcode} + OP_POP) + n * OP_POP;
+        }
+        case InstructionKind::nullop:
+            // CALLER POP CALLER POP ...
+            return stack_limit * (bytecode{opcode} + OP_POP);
+        case InstructionKind::nop:
+            // JUMPDEST JUMPDEST ...
+            return stack_limit * 2 * bytecode{opcode};
+        case InstructionKind::unop:
+            // DUP1 NOT NOT ... POP
+            return OP_DUP1 + stack_limit * 2 * bytecode{opcode} + OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 ADD DUP1 ADD DUP1 ADD ... POP
+            return OP_DUP1 + (stack_limit - 1) * (OP_DUP1 + bytecode{opcode}) + OP_POP;
+        default:
+            break;
+        }
+        break;
+    case Mode::full_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            // PUSH1 PUSH1 PUSH1 ... POP POP POP ...
+            return stack_limit * push(opcode, {}) + stack_limit * OP_POP;
+        case InstructionKind::dup:
+        {
+            const auto n = opcode - OP_DUP1;
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * bytecode{opcode} + stack_limit * OP_POP;
+        }
+        case InstructionKind::nullop:
+            // CALLER CALLER ... POP POP ...
+            return stack_limit * opcode + stack_limit * OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 DUP1 ... ADD ADD ADD ... POP
+            return stack_limit * OP_DUP1 + (stack_limit - 1) * opcode + OP_POP;
+        default:
+            break;
+        }
+        break;
+    }
+
+    return {};
+}
+
+/// Generates a benchmark loop with given inner code.
+///
+/// This generates do-while loop with 255 iterations and it starts with PUSH1 of 255 as the loop
+/// counter. The while check is done as `(counter += -1) != 0`. The SUB is avoided because it
+/// consumes arguments in unnatural order and additional SWAP would be required.
+///
+/// The loop counter stays on the stack top. The inner code is allowed to duplicate it, but must not
+/// modify it.
+bytecode generate_loop_v1(const bytecode& inner_code)
+{
+    return push(255) + OP_JUMPDEST + inner_code +
+           push("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") + OP_ADD +
+           OP_DUP1 + push(2) + OP_JUMPI;
+}
+
+/// Generates a benchmark loop with given inner code.
+///
+/// This is improved variant of v1. It has exactly the same instructions and consumes the same
+/// amount of gas, but according to performed benchmarks (see "loop_v1" and "loop_v2") it runs
+/// faster. And we want the lowest possible loop overhead.
+/// The change is to set the loop counter to -255 and check `(counter += 1) != 0`.
+bytecode generate_loop_v2(const bytecode& inner_code)
+{
+    return push("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01") + OP_JUMPDEST +
```

Comment:
Following the comment from above, since here we can not use `uint256_t{}`, maybe a better options is to:
```
auto counter = push("ff...");
return counter + inner_code ... + push(counter.size()) + OP_JUMPI;
```

In both loop generators.